sql/core/src/main/scala/org/apache/spark/sql/execution/stat/ContingencyTable.scala

```diff
@@ -0,0 +1,38 @@
+package org.apache.spark.sql.execution.stat
+
+import org.apache.spark.sql.{Row, DataFrame}
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.functions._
+
+
+private[sql] object ContingencyTable {
+
+  /** Generate a table of frequencies for the elements of two columns. */
+  private[sql] def crossTabulate(df: DataFrame, col1: String, col2: String): DataFrame = {
+    val tableName = s"${col1}_$col2"
+    val distinctVals = df.select(countDistinct(col1), countDistinct(col2)).collect().head
```

Comment:
This implementation triggers multiple jobs. I'm thinking about the following approach:

1. get distinct values from col2 and create a value-to-index map
2. aggregate by col1. for each value in col1, generate a Row object and fill in counts
3. assign table schema